src/components/media/MediasSelectModal.jsx

```diff
@@ -233,23 +240,36 @@ const MediasSelectModal = ({
                 {files &&
                   files
                     .filter(({ data }) => filteredMedias.includes(data?.name))
-                    .map(({ data, isLoading }, mediaItemIndex) => (
+                    .map(({ data, isLoading }) => (
                       <Skeleton
                         w="100%"
                         h={isLoading ? "4.5rem" : "fit-content"}
                         isLoaded={!isLoading}
                         key={data.name}
                       >
-                        <MediaCard
-                          type={mediaRoom}
-                          media={data}
-                          mediaItemIndex={mediaItemIndex}
-                          onClick={() => onMediaSelect(data)}
-                          isSelected={
-                            data.name === watch("selectedMedia")?.name
-                          }
-                          showSettings={false}
-                        />
+                        {pluralMediaLabel === "images" ? (
+                          <ImagePreviewCard
+                            name={data.name}
+                            addedTime={data.addedTime}
+                            mediaUrl={data.mediaUrl}
+                            isSelected={
+                              data.name === watch("selectedMedia")?.name
```

Comment:
Would this work as intended if two different albums have the same filename, and the user selected one of them before navigating to the other?

Reply:
Oh wow good catch - there's a bug present here that prevents the images in the folder from loading when an existing image is already selected? I've opted to remove the currently selected file when the user swaps directories to remove this issue in https://github.com/isomerpages/isomercms-frontend/pull/1678/commits/d8c29dd067e5ec5d20073eb1a8ceb12b820b8798